src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -774,12 +774,18 @@ def list_vm_ip_addresses(cmd, resource_group_name=None, vm_name=None):
                 network_info['privateIpAddresses'].append(ip_configuration.private_ip_address)
                 if ip_configuration.public_ip_address:
                     public_ip_address = ip_address_lookup[ip_configuration.public_ip_address.id]
-                    network_info['publicIpAddresses'].append({
+
+                    public_ip_addr_info = {
                         'id': public_ip_address.id,
                         'name': public_ip_address.name,
                         'ipAddress': public_ip_address.ip_address,
                         'ipAllocationMethod': public_ip_address.public_ip_allocation_method
-                    })
+                    }
+
+                    if hasattr(public_ip_address, 'zones'):
+                        public_ip_addr_info['zones'] = public_ip_address.zones if public_ip_address.zones else []
```

Comment:
Please change the field name to `zone` as public ip is never a cross-zone resource. `zones` is service end naming defect.

Reply:
Okay. Should I add logic to handle the event where there are multiple zones returned in the list. Or should I assume it would never happen and just always return the first element if it exists?

Because it looks like from the [swagger](https://github.com/Azure/azure-rest-api-specs/blob/32b0d873aa851d456dfde7d6ba1d89ff33f897f0/specification/network/resource-manager/Microsoft.Network/stable/2018-05-01/publicIpAddress.json#L403-L409) / sdk,  a public ip address can have multiple availability zones.

Reply:
nah, it will never happen